cristin-commons/src/main/java/no/unit/nva/cristin/model/QueryParameterKey.java

```diff
@@ -0,0 +1,183 @@
+package no.unit.nva.cristin.model;
+
+import static java.util.Objects.nonNull;
+import static no.unit.nva.cristin.common.ErrorMessages.ALPHANUMERIC_CHARACTERS_DASH_COMMA_PERIOD_AND_WHITESPACE;
+import static no.unit.nva.cristin.common.ErrorMessages.ERROR_MESSAGE_INVALID_NUMBER;
+import static no.unit.nva.cristin.common.ErrorMessages.ERROR_MESSAGE_INVALID_PATH_PARAMETER_FOR_ID_FOUR_NUMBERS;
+import static no.unit.nva.cristin.common.ErrorMessages.ERROR_MESSAGE_INVALID_VALUE;
+import static no.unit.nva.cristin.common.ErrorMessages.INVALID_URI_MESSAGE;
+import static no.unit.nva.cristin.model.Constants.CRISTIN_PER_PAGE_PARAM;
+import static no.unit.nva.cristin.model.Constants.CRISTIN_QUERY_NAME_PARAM;
+import static no.unit.nva.cristin.model.Constants.PARENT_UNIT_ID;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_DATE;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_LANGUAGE;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_NON_EMPTY;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_NUMBER;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_TITLE;
+import static no.unit.nva.cristin.model.Constants.PATTERN_IS_URL;
+import static no.unit.nva.cristin.model.Constants.PROJECTS_PATH;
+import static no.unit.nva.cristin.model.Constants.PROJECT_PATH_NVA;
+import static no.unit.nva.cristin.model.Constants.QUERY_PARAMETER_LANGUAGE;
+import static no.unit.nva.cristin.model.QueryParameterKey.QueryParameterConstant.ERROR_MESSAGE_INVALID_CHARACTERS;
+import static no.unit.nva.model.Organization.ORGANIZATION_IDENTIFIER_PATTERN;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public enum QueryParameterKey {
+    INVALID(null),
+    PATH_IDENTITY(JsonPropertyNames.IDENTIFIER, null, PATTERN_IS_NON_EMPTY),
+    PATH_ORGANISATION(PARENT_UNIT_ID,
+        JsonPropertyNames.ORGANIZATION,
+        ORGANIZATION_IDENTIFIER_PATTERN,
+        ERROR_MESSAGE_INVALID_PATH_PARAMETER_FOR_ID_FOUR_NUMBERS,
+        true),
+    PATH_PROJECT(PROJECTS_PATH, PROJECT_PATH_NVA, PATTERN_IS_NUMBER),
+    APPROVAL_REFERENCE_ID(JsonPropertyNames.PROJECT_APPROVAL_REFERENCE_ID),
+    APPROVED_BY(JsonPropertyNames.PROJECT_APPROVED_BY),
+    BIOBANK(JsonPropertyNames.BIOBANK_ID),
+    FUNDING(JsonPropertyNames.FUNDING),
+    FUNDING_SOURCE(JsonPropertyNames.FUNDING_SOURCE),
+    GRANT_ID("project_code", null, PATTERN_IS_NUMBER),
```

Comment:
have no nva name?

Reply:
Jo, for å ikke gjenta det samme navnet flere ganger (når nva og cristin bruker samme parameternavn) sender jeg med null og så setter konstruktøren verdien på begge.

Om du mener det er en dårlig måte å gjøre det på skifter jeg.

Reply:
Om nvakey == cristinkey -> nvakey = cristinkey